covert/idstore.py

```diff
@@ -0,0 +1,93 @@
+from covert import passphrase, pubkey
+from covert.blockstream import encrypt_file, decrypt_file
+from covert.archive import Archive
+from pathlib import Path
+import mmap
+import subprocess
+import os
+
+def idfilename():
+  # TODO: Use xdg paths such as ~/.config/covert/idstore on unix systems
+  return Path.home() / ".covert" / "idstore"
+
+def create(pwhash, idstore=None):
+  a = Archive()
+  a.index["I"] = idstore or {}
+  # Encrypt in RAM...
+  out = b"".join(b for b in encrypt_file((False, [pwhash], [], []), a.encode, a))
+  fn = idfilename()
+  confdir = idfilename().parent
+  if not confdir.exists():
+    confdir.mkdir(parents=True)
+    if os.name == "posix":
```

Comment:
I would remove 'if os.name == "x": function(object)' statements, and replace them with a very thin layer: os_util.chmod(confdir, only_the_user=True).

os_util.chmod function would check operating system / platform specifics.